src/details/ArborX_DetailsDistributedTreeUtils.hpp

```diff
@@ -331,13 +331,15 @@ void communicateResultsBack(MPI_Comm comm, ExecutionSpace const &space,
   }
 }
 
-template <typename ExecutionSpace, typename BottomTree, typename Predicates,
-          typename Callback, typename RanksTo, typename Offset, typename Values,
-          typename Ranks>
+template <
+    typename ExecutionSpace, typename BottomTree, typename Predicates,
+    typename Callback, typename RanksTo, typename Offset, typename Values,
+    typename Ranks = Kokkos::View<int *, typename BottomTree::memory_space>>
 void forwardQueriesAndCommunicateResults(
     MPI_Comm comm, ExecutionSpace const &space, BottomTree const &bottom_tree,
     Predicates const &predicates, Callback const &callback,
-    RanksTo const &ranks_to, Offset &offset, Values &values, Ranks &ranks)
+    RanksTo const &ranks_to, Offset &offset, Values &values,
+    Ranks *ranks_ptr = nullptr)
```

Comment:
I would prefer using `std::optional` here
```suggestion
    std::optional<Ranks> ranks = {})
```

Reply:
Not doing it. This argument will be gone anyway in #1075.